rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +605,121 @@ def _query_lending_balances(
 
         return balances
 
+    def query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> bool:
+        """Queries Binance lending interest history, transforms it into `HistoryBaseEntry` objects and save it in the database.
+
+        "DAILY" -> flexible savings.
+        "CUSTOMIZED_FIXED" -> fixed savings.
+        "ACTIVITY" -> fixed savings but as a result of token partners.
+
+        Lending Interest History Documentation:
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
+
+        May raise:
+        - RemoteError
+        - BinancePermissionError
+        """  # noqa: E501
+        ranges = DBQueryRanges(self.db)
+        range_query_name = f'{self.location}_lending_history_{self.name}'
+        ranges_to_query = ranges.get_location_query_ranges(
+            cursor=cursor,
+            location_string=range_query_name,
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        for query_start_ts, query_end_ts in ranges_to_query:
+            events = []
+            for lending_type in ('DAILY', 'ACTIVITY', 'CUSTOMIZED_FIXED'):
+                try:
+                    response = self._api_query_list_within_time_delta(
+                        api_type='sapi',
+                        start_ts=query_start_ts,
+                        time_delta=BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS,
+                        end_ts=query_end_ts,
+                        method='lending/union/interestHistory',
+                        # size determines the length of the data returned,
+                        # 100 is the maximum value possible
+                        additional_options={'lendingType': lending_type, 'size': 100},
+                    )
+                except (RemoteError, BinancePermissionError) as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to query binance lending interest history between '
+                        f'{query_start_ts} and {query_end_ts}. {str(e)}',
+                    )
+                    return True
+
+                for entry in response:
+                    try:
+                        interest_received = deserialize_asset_amount(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        timestamp = TimestampMS(deserialize_timestamp_from_intms(entry['time']) * 1000)  # noqa: E501
```

Comment:
What are you doing here? This is completely wrong.

`entry['time']` is a timestam in MS. You are passing it to `deserialize_timestamp_from_intms` which takes it and divides it by `1000` to get the normal unix timestamp ... and then you multiply it again by 1000? Wtf? Just cast it normally and directly.